runelite-client/src/main/java/net/runelite/client/plugins/performance/PerformancePlugin.java

```diff
@@ -95,12 +118,26 @@ protected void startUp() throws Exception
 		overlayManager.add(overlay);
 		drawManager.registerEveryFrameListener(drawListener);
 		drawListener.reloadConfig();
+		pingExecutorService.scheduleAtFixedRate(this::getPingToCurrentWorld, 5, 5, TimeUnit.SECONDS);
 	}
 
 	@Override
 	protected void shutDown() throws Exception
 	{
 		overlayManager.remove(overlay);
 		drawManager.unregisterEveryFrameListener(drawListener);
+		pingExecutorService.shutdown();
+	}
+
+	private void getPingToCurrentWorld()
+	{
+		if (client.getGameState().equals(GameState.LOGGED_IN) && performanceConfig.drawPing())
+		{
+			ping = Ping.ping(String.format("oldschool%d.runescape.com", client.getWorld() - 300));
```

Comment:
This might potentially break at some point if JaGeX decides to hosts worlds that are not available trough the `runescape.com` subdomain (_highly unlikely, but who knows_), is there a way to retrieve the host of the current server you are connected to?